api/pool.go

```diff
@@ -158,6 +159,16 @@ func removePoolHandler(w http.ResponseWriter, r *http.Request, t auth.Token) (er
 		return permission.ErrUnauthorized
 	}
 	poolName := r.URL.Query().Get(":name")
+	filter := &app.Filter{}
+	filter.Pool = poolName
+	contexts := permission.ContextsForPermission(t, permission.PermAppRead)
+	apps, err := app.List(appFilterByContext(contexts, filter))
```

Comment:
I think we could use a empty slice of contexts here (instead of getting the contexts from the user token) so we also take into account apps that the user has no permission on.